core/src/test/java/com/linecorp/armeria/common/util/CompositeExceptionTest.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.util;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Test;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.internal.testing.AnticipatedException;
+
+class CompositeExceptionTest {
+
+    private static final String separator = System.getProperty("line.separator");
+
+    @Test
+    void verboseExceptionEnabledTest() {
+        final IllegalStateException ex1 = new IllegalStateException();
+        final IllegalArgumentException ex2 = new IllegalArgumentException();
+        final CompositeException compositeException =
+                new CompositeException(ImmutableList.of(ex1, ex2), Sampler.always());
+        final long separatedStacktraceLength = stackTraceLength(compositeException);
+
+        final int sumStackTraceOutputLength = ex1.getStackTrace().length + ex2.getStackTrace().length;
+        // if verboseException option enabled, output full stacktrace
+        // this case is occurred 2 exceptions (ex1 stacktrace length + ex2 stacktrace length)
+        assertThat(separatedStacktraceLength).isEqualTo(sumStackTraceOutputLength);
+    }
+
+    @Test
+    void verboseExceptionDisabledTest() {
+        final IllegalStateException ex1 = new IllegalStateException();
+        final IllegalArgumentException ex2 = new IllegalArgumentException();
+        final CompositeException compositeException =
+                new CompositeException(ImmutableList.of(ex1, ex2), Sampler.never());
+        final long separatedStacktraceLength = stackTraceLength(compositeException);
+
+        // if verboseException option disabled, max output stacktrace is
+        // CompositeException.DEFAULT_MAX_NUM_STACK_TRACES.
+        // this case is occurred 2 exceptions (20 * 2)
+        assertThat(separatedStacktraceLength).isEqualTo(
+                CompositeException.DEFAULT_MAX_NUM_STACK_TRACES * 2);
```

Comment:
This test could be broken if the original stack trace is less than `DEFAULT_MAX_NUM_STACK_TRACES`.
How about manually setting stackframes longer than 20 like we added in `verboseExceptionEnabledTestMultipleContainsCustomStackTraces`?

Reply:
Sounds good :)

It would also be good to manually change the `verboseExceptionDisabledTestMultipleContainsCustomStackTraces` method similar to the code above.